### PR TITLE
clang-tidy: range for loop conversions

### PR DESCRIPTION
**Pull request by neheb**
Found with modernize-loop-convert

Ran through git clang-format.

Also removed several questionable loops and replaced with simpler
algorithms.

Signed-off-by: Rosen Penev <rosenp@gmail.com>

### CODE REVIEW

File: src/epsimage.cpp [hunk 1]
```diff
@@ -1175,9 +1171,13 @@ namespace Exiv2
         }
         // check for all possible (DOS) EPS signatures
         bool matched = (memcmp(buf.pData_, dosEpsSignature.data(), dosEpsSignature.size()) == 0);
-        for (size_t i = 0; !matched && i < (sizeof epsFirstLine) / (sizeof *epsFirstLine); i++) {
-            matched = (memcmp(buf.pData_, epsFirstLine[i].data(), epsFirstLine[i].size()) == 0);
-        }
+        if (!matched)
+            for (auto&& eps : epsFirstLine) {
+                if (memcmp(buf.pData_, eps.data(), eps.size()) == 0) {
```
Comment (neheb) [inline src/epsimage.cpp:1176]: Interestingly enough std::any_of or std::find can't be used here. No idea why.

**Review by hassec — CHANGES_REQUESTED** [review]

File: samples/easyaccess-test.cpp [hunk 2]
```diff
@@ -87,9 +87,9 @@ try {
     image->readMetadata();
     Exiv2::ExifData& ed = image->exifData();
 
-    for (unsigned int i = 0; i < EXV_COUNTOF(easyAccess); ++i) {
-        auto pos = easyAccess[i].findFct_(ed);
-        std::cout << std::setw(21) << std::left << easyAccess[i].label_;
+    for (auto&& easyAcces : easyAccess) {
```
[the remark this answers is not in the document]
Comment (neheb) [reply samples/easyaccess-test.cpp:90]: The name was automated. Will change.

File: samples/exifdata.cpp [hunk 3]
```diff
@@ -43,11 +43,7 @@ void syntax(const char* argv[],format_t& formats)
 
 size_t formatInit(Exiv2::ExifData& exifData)
 {
-	size_t result = 0;
-	for (auto i = exifData.begin(); i != exifData.end() ; ++i) {
-		result ++ ;
-	}
-	return result ;
+    return std::distance(exifData.begin(), exifData.end());
```
[the remark this answers is not in the document]
Comment (neheb) [reply samples/exifdata.cpp:46]: Sounds right.

**Review by piponazo — DISMISSED** [review]
LGTM, thanks for this nice contribution!

Could you please rebase your branch in top of `main`? It looks like there are some compiler errors on the appveyor builds.

**Comment by neheb** [comment]
Rebased.

**Review by piponazo — APPROVED** [review]

**Pull request merged**